### ISSUE TITLE
Question about the absence of attention mask in parallel decoding

### ISSUE THREAD

**Issue by Bitterdhg** [hp-l33/ARPG#2]
Hi @hp-l33 , I noticed that the attention mask is not used [[this line]](https://github.com/hp-l33/ARPG/blob/6262b0ddda6349bb7ecf40d7f513f7c500e777c3/models/arpg.py#L172). It is OK if ``num_pred``  is 1. 

But when ``num_pred`` > 1, the attention pattern is inconsistent between training and inference.

I am curious as to whether this is a bug or an intended feature within the APRG framework. 
Because RandAR still uses attention mask in parallel decoding  [[this line]](https://github.com/ziqipang/RandAR/blob/9f4707bb88127b1125c4c558446b3c216ddd5182/RandAR/model/randar_gpt.py#L442).

**Comment by hp-l33** [comment]
Thank you for your interest in our work.

This is indeed a feature of our method. Since our method can handle existing token sequences of arbitrary lengths in a random order, our parallel decoding actually treats the decoding of tokens at multiple positions as independent processes that do not interfere with each other. Therefore, there is no need for masking during inference. It should be noted that our query is data-independent, so not adding a mask during inference will not lead to a inconsistent between training and inference. You can also understand it this way: the reason we add a causal mask during training in our method is to construct variable-length token sequences.

**Comment by hp-l33** [comment]
Hi @Bitterdhg .

I hope my previous reply was able to answer your questions.

If you have any further questions, please feel free to consult me. I believe that communication will enable me to refine the expressions in the paper more effectively during subsequent updates, making it more accessible to readers.

Best regards.
